dbos/dbos.py

```diff
@@ -138,16 +171,64 @@ def send_temp_workflow(
         temp_send_wf = _workflow_wrapper(self, send_temp_workflow)
         set_dbos_func_name(send_temp_workflow, TEMP_SEND_WF_NAME)
         self._register_wf_function(TEMP_SEND_WF_NAME, temp_send_wf)
+
+    @property
+    def sys_db(self) -> SystemDatabase:
+        if self._sys_db is None:
+            self.launch()
+        if self._sys_db is None:
+            raise DBOSException("System database not initialized")
+        rv: SystemDatabase = self._sys_db
+        return rv
+
+    @property
+    def app_db(self) -> ApplicationDatabase:
+        if self._app_db is None:
+            self.launch()
+        if self._app_db is None:
+            raise DBOSException("Application database not initialized")
+        rv: ApplicationDatabase = self._app_db
+        return rv
+
+    @property
+    def admin_server(self) -> AdminServer:
+        if self._admin_server is None:
+            self.launch()
+        if self._admin_server is None:
+            raise DBOSException("Admin server is not initialized")
+        rv: AdminServer = self._admin_server
+        return rv
+
+    def launch(self) -> None:
```

Comment:
I also don't understand the difference between `__init__` and `launch`.  When do we need the latter?

Reply:
I still don't like the idea that the thing is running immediately after it is constructed and before the handlers are registered.  So it is split up.  We may find reasons to adjust as we work on demo apps, as we discussed yesterday.

When we do serverless, this (or some variant thereof) will fix a lot of trouble and make a lot of sense.

Reply:
What does it mean to be launched but not initialized?

Reply:
This will be clearer in the new lifecycle